### PR TITLE
Docs: adding wrapper modularization docs

### PR DESCRIPTION
**Pull request by kirszenbaum**
This PR gathers docs updates related to the new wrapper modularization features:
- [x] Adds info on the new registering functions (re: #8846)
- [x] Adds info on using modules with the React wrapper (re: #8854) 
- [x] Adds info on using modules with the Vue wrapper (re: #8861)
- [x] Adds info on using modules with the Angular wrapper (re: #8862)
- [x] Reshuffles content in the Modules topic a bit
- [x] Updates all wrapper code samples with the new `registerAllModules()` function (re: [Slack](https://handsoncode.slack.com/archives/C4PV4S33Q/p1634209383002900?thread_ts=1634144473.003800&cid=C4PV4S33Q)) **(but doesn't update CodeSandbox demos on [this](https://handsontable.com/docs/react-hot-column/) page**)

[skip changelog]

### CODE REVIEW

File: docs/next/guides/building-and-testing/modules.md [hunk 1]
```diff
@@ -143,129 +145,483 @@ Elements to be imported manually on demand:
...
+import { ColumnSorting } from 'handsontable/plugins/columnSorting';
+import { Comments } from 'handsontable/plugins/comments';
+import { ContextMenu } from 'handsontable/plugins/contextMenu';
+import { CopyPaste } from 'handsontable/plugins/copyPaste';
+import { CustomBorders } from 'handsontable/plugins/customBorders';
+import { DragToScroll } from 'handsontable/plugins/dragToScroll';
+import { DropdownMenu } from 'handsontable/plugins/dropdownMenu';
+import { ExportFile } from 'handsontable/plugins/exportFile';
+import { Filters } from 'handsontable/plugins/filters';
+import { HeaderTooltips } from 'handsontable/plugins/headerTooltips';
```
Comment (budnix) [inline docs/next/guides/building-and-testing/modules.md:209]: The import can be removed as the plugin was removed in [9.0.0](https://github.com/handsontable/handsontable/releases/tag/9.0.0).

File: docs/next/guides/building-and-testing/modules.md [hunk 2]
```diff
@@ -143,129 +145,483 @@ Elements to be imported manually on demand:
...
+import { ManualColumnFreeze } from 'handsontable/plugins/manualColumnFreeze';
+import { ManualColumnMove } from 'handsontable/plugins/manualColumnMove';
+import { ManualColumnResize } from 'handsontable/plugins/manualColumnResize';
+import { ManualRowMove } from 'handsontable/plugins/manualRowMove';
+import { ManualRowResize } from 'handsontable/plugins/manualRowResize';
+import { MergeCells } from 'handsontable/plugins/mergeCells';
+import { MultipleSelectionHandles } from 'handsontable/plugins/multipleSelectionHandles';
+import { NestedHeaders } from 'handsontable/plugins/nestedHeaders';
+import { NestedRows } from 'handsontable/plugins/nestedRows';
+import { ObserveChanges } from 'handsontable/plugins/observeChanges';
```
Comment (budnix) [inline docs/next/guides/building-and-testing/modules.md:221]: The import can be removed as the plugin was removed in [9.0.0](https://github.com/handsontable/handsontable/releases/tag/9.0.0).

File: docs/next/guides/building-and-testing/modules.md [hunk 3]
```diff
@@ -143,129 +145,483 @@ Elements to be imported manually on demand:
...
+  MultipleSelectionHandles,
+  NestedHeaders,
+  NestedRows,
+  ObserveChanges,
+  PersistentState,
+  Search,
+  TouchScroll,
+  TrimRows,
+  UndoRedo,
+  registerPlugin,
```
Comment (budnix) [inline docs/next/guides/building-and-testing/modules.md:344]:
```suggestion
  AutoColumnSize,
  AutoRowSize,
  Autofill,
  BasePlugin,
  BindRowsWithHeaders,
  CollapsibleColumns,
  ColumnSorting,
  ColumnSummary,
  Comments,
  ContextMenu,
  CopyPaste,
  CustomBorders,
  DragToScroll,
  DropdownMenu,
  ExportFile,
  Filters,
  Formulas,
  HiddenColumns,
  HiddenRows,
  ManualColumnFreeze,
  ManualColumnMove,
  ManualColumnResize,
  ManualRowMove,
  ManualRowResize,
  MergeCells,
  MultiColumnSorting,
  MultipleSelectionHandles,
  NestedHeaders,
  NestedRows,
  PersistentState,
  Search,
  TouchScroll,
  TrimRows,
  UndoRedo,
  registerPlugin,
```

File: docs/next/guides/building-and-testing/modules.md [hunk 4]
```diff
@@ -143,129 +145,483 @@ Elements to be imported manually on demand:
...
+// optionally, import registering functions
+// that let you quickly register all modules at once
+import {
+  registerAllEditors,
+  registerAllRenderers,
+  registerAllValidators,
+  registerAllCellTypes,
+  registerAllPlugins,
+  registerAllModules,
+  } from 'handsontable/registry'
```
Comment (budnix) [inline docs/next/guides/building-and-testing/modules.md:377]:
```suggestion
import {
  registerAllEditors,
  registerAllRenderers,
  registerAllValidators,
  registerAllCellTypes,
  registerAllPlugins,
  registerAllModules,
} from 'handsontable/registry'
```
Fix formatting :)

File: docs/next/guides/building-and-testing/modules.md [hunk 5]
```diff
@@ -143,129 +145,483 @@ Elements to be imported manually on demand:
...
+registerPlugin(NestedHeaders);
+registerPlugin(NestedRows);
+registerPlugin(ObserveChanges);
+registerPlugin(PersistentState);
+registerPlugin(Search);
+registerPlugin(TouchScroll);
+registerPlugin(TrimRows);
+registerPlugin(UndoRedo);
+// or, register all plugins at once
+registerAllPlugins();
```
Comment (budnix) [inline docs/next/guides/building-and-testing/modules.md:480]: It's a great idea to include these shortcut functions here along with the single ones 💪

File: docs/next/guides/building-and-testing/modules.md [hunk 6]
```diff
@@ -143,129 +145,483 @@ Elements to be imported manually on demand:
...
+  PasswordEditor,
+  SelectEditor,
+  TextEditor,
+} from 'handsontable/editors';
+
+// import the `registerAllEditors()` function
+import { registerAllEditors } from 'handsontable/registry';
+
+// register all editors
+registerAllEditors();
```
Comment (budnix) [inline docs/next/guides/building-and-testing/modules.md:556]:
```suggestion
// import the `registerAllEditors()` function
import { registerAllEditors } from 'handsontable/registry';

// register all editors
registerAllEditors();
```
There is no need to import the editors when you're using the `registerAllEditors` function.

File: docs/next/guides/building-and-testing/modules.md [hunk 7]
```diff
@@ -143,129 +145,483 @@ Elements to be imported manually on demand:
...
+  passwordRenderer,
+  textRenderer,
+} from 'handsontable/renderers';
+
+// import the `registerAllRenderers()` function
+import { registerAllRenderers } from 'handsontable/registry';
+
+// register all renderers
+registerAllRenderers();
+```
```
Comment (budnix) [inline docs/next/guides/building-and-testing/modules.md:623]:
```suggestion
```js
// import the `registerAllRenderers()` function
import { registerAllRenderers } from 'handsontable/registry';

// register all renderers
registerAllRenderers();
```
```

File: docs/next/guides/building-and-testing/modules.md [hunk 8]
```diff
@@ -309,6 +665,27 @@ new Handsontable(container, {
...
+  numericValidator,
+  timeValidator,
+} from 'handsontable/validators';
+
+// import the `registerAllValidators()` function
+import { registerAllValidators } from 'handsontable/registry';
+
+// register all validators
+registerAllValidators();
+```
```
Comment (budnix) [inline docs/next/guides/building-and-testing/modules.md:687]:
```suggestion
```js
// import the `registerAllValidators()` function
import { registerAllValidators } from 'handsontable/registry';

// register all validators
registerAllValidators();
```
```

File: docs/next/guides/building-and-testing/modules.md [hunk 9]
```diff
@@ -351,231 +728,77 @@ new Handsontable(container, {
...
+  AutocompleteCellType,
+  CheckboxCellType,
+  DateCellType,
+  DropdownCellType,
+  HandsontableCellType,
+  NumericCellType,
+  PasswordCellType,
+  TextCellType,
+  TimeCellType,
+} from 'handsontable/cellTypes';
```
Comment (budnix) [inline docs/next/guides/building-and-testing/modules.md:748]:
```suggestion
```

File: docs/next/guides/building-and-testing/modules.md [hunk 10]
```diff
@@ -615,112 +838,311 @@ import {
...
+new Handsontable(container, {
+  type: 'date',
+});
+```
+
+## Using modules with wrappers
+
+You can use modules with Handsontable's [React](@/guides/integrate-with-react/react-installation.md), [Angular](@/guides/integrate-with-angular/angular-installation.md), and [Vue 2](@/guides/integrate-with-vue/vue-installation.md) wrappers.
+
+### Using modules with React
```
Comment (budnix) [inline docs/next/guides/building-and-testing/modules.md:927]: I don't know but maybe we could link or move that part of the documentation about modules to the framework part? 🤔

**Review by budnix — CHANGES_REQUESTED** [review]
I left my suggestions in the code.

File: docs/next/guides/building-and-testing/modules.md (hunk 1 above)
Comment (kirszenbaum) [reply docs/next/guides/building-and-testing/modules.md]: Done: ca35bb7

File: docs/next/guides/building-and-testing/modules.md (hunk 2 above)
Comment (kirszenbaum) [reply docs/next/guides/building-and-testing/modules.md]: Done: ca35bb7

File: docs/next/guides/building-and-testing/modules.md (hunk 3 above)
Comment (kirszenbaum) [reply docs/next/guides/building-and-testing/modules.md]: Done: ca35bb7

File: docs/next/guides/building-and-testing/modules.md (hunk 10 above)
Comment (kirszenbaum) [reply docs/next/guides/building-and-testing/modules.md]: Yeah, I was thinking about that. I think you're right: framework users probably look mainly at the "Integrate with React/Vue/Angular" section (ofc we don't have any analytics data to support that assumption, though), so info about modules should definitely be there. And, we can link to those sections from the "Modules" topic.

Separate topics also nicely display in search:


Here's the change: a5133d1

**Review by budnix — CHANGES_REQUESTED** [review]
@kirszenbaum Can you check what happens in the http://localhost:8080/docs/next/modules/#importing-renderers?

image

File: docs/next/guides/building-and-testing/modules.md [hunk 11]
```diff
@@ -351,231 +684,64 @@ new Handsontable(container, {
...
-## Importing locales
+#### Importing all cell types at once
 
-Importing locales works slightly different than in case of other elements. Let's try adding the `pl-PL` locale.
+You can quickly import and register all cell types at once:
 
-Start with importing the base and the language code:
+```js
+
+// import the `registerAllCellTypes()` function
```
Comment (budnix) [inline docs/next/guides/building-and-testing/modules.md:693]:
```suggestion
```js
// import the `registerAllCellTypes()` function
```
The suggestion removes the redundant empty line.

File: docs/next/guides/building-and-testing/modules.md [hunk 12]
```diff
@@ -351,231 +684,64 @@ new Handsontable(container, {
...
+import { registerAllCellTypes } from 'handsontable/registry';
```
Comment (budnix) [inline docs/next/guides/building-and-testing/modules.md:694]:
```suggestion
// import the `registerAllCellTypes()` function
import { registerAllCellTypes } from 'handsontable/registry';
```
The suggestion removes the redundant empty line.

**Comment by budnix** [comment]
The import of individual plugins is not necessary here. Can I ask you to remove update that code snippet?
2021-11-4 o 10 22 54

File: docs/next/guides/integrate-with-angular/angular-modules.md [hunk 13]
```diff
@@ -0,0 +1,95 @@
...
+
+## Overview
+
+To reduce the size of your app, you can use Handsontable by importing individual [modules](@/guides/building-and-testing/modules.md).
+
+## Using modules with Angular
+
+To use modules with Handsontable's [Angular wrapper](@/guides/integrate-with-angular/angular-installation.md), follow the steps below:
+
+### Step 1: Import the `Base` module
```
Comment (budnix) [inline docs/next/guides/integrate-with-angular/angular-modules.md:20]:
```suggestion
### Step 1: Import the `handsontable/base` module
```
I think that the `Base` name here can be too generic for new developers. What do you think about it to call it the same as the package is named?

File: docs/next/guides/integrate-with-angular/angular-modules.md [hunk 14]
```diff
@@ -0,0 +1,95 @@
...
+In the entry point file of your application, import Handsontable's `Base` module:
```
Comment (budnix) [inline docs/next/guides/integrate-with-angular/angular-modules.md:21]:
```suggestion
In the entry point file of your application, import Handsontable's `handsontable/base` module:
```

**Review by budnix — CHANGES_REQUESTED** [review]

File: docs/next/guides/integrate-with-angular/angular-modules.md [hunk 15]
```diff
@@ -0,0 +1,95 @@
...
+  import { NgModule } from '@angular/core';
+  import { AppComponent } from './app.component';
+  import { HotTableModule } from '@handsontable/angular';
+
+  // import Handsontable's `Base` module
+  import Handsontable from 'handsontable/base';
+  ```
+
+### Step 2: Import modules and their registering functions
+Import the required modules (for the list of Handsontable modules, see the [modules cheatsheet](@/guides/building-and-testing/modules.md#modules-cheatsheet)).
```
Comment (budnix) [inline docs/next/guides/integrate-with-angular/angular-modules.md:33]: WDYT to rephrase the sentence "Import the required modules..." to something like "Import the modules that you want to use..."? "required" sounds to "required" for me 🤔

**Comment by kirszenbaum** [comment]
> @kirszenbaum Can you check what happens in the http://localhost:8080/docs/next/modules/#importing-renderers?
> 
> image
> 
> EDIT: The same happened in http://localhost:8080/docs/next/modules/#importing-renderers.

@budnix - my bad! Fixed here: 60a2024

**Comment by kirszenbaum** [comment]
> The import of individual plugins is not necessary here. Can I ask you to remove update that code snippet? 2021-11-4 o 10 22 54

@budnix - done: 85099d0

File: docs/next/guides/integrate-with-angular/angular-modules.md (hunk 14 above)
Comment (kirszenbaum) [reply docs/next/guides/integrate-with-angular/angular-modules.md]: Done here: 75c8d4e

File: docs/next/guides/integrate-with-angular/angular-modules.md (hunk 15 above)
Comment (kirszenbaum) [reply docs/next/guides/integrate-with-angular/angular-modules.md]: You're right, "required" is ambiguous here. Done: 7841591

**Review by budnix — APPROVED** [review]
🏅  Thanks! There is a problem with wrapper demos but based on this branch it's already fixed and accepted in the [#8915](https://github.com/handsontable/handsontable/pull/8915). As you merge the changes to the `develop` branch I will merge mine.

**Pull request merged**